### PR TITLE
Fixes to Jail square sizing on GamePanel.

### PR DESCRIPTION
**Pull request by ghost**
Introduction of MiGLayout. Updates to many layouts to use MiG instead. Set minimum sizes for Chessframe to prevent weird sizing issues below the minimum.

### CODE REVIEW

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java [hunk 1]
```diff
@@ -17,23 +19,51 @@
...
-    private final int mJailDimension;
+    private final int JAIL_WIDTH = 4;
     private final List<SquareJLabel> mSquareLabels;
+    private final BoardSize mJailSize;
 
     public JailPanel(int totalPieces) {
         super(false);
 
-        mJailDimension = (totalPieces / 4) + (totalPieces % 4);
+        if(totalPieces % 4 == 0) {
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java]: `totalPieces % JAIL_WIDTH` ?

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/TeamStatusPanel.java [hunk 2]
```diff
@@ -66,59 +65,41 @@ public void updateStatus(int activeTeamId, Status status) {
     }
 
     private void initComponents() {
-        GridBagConstraints constraints = new GridBagConstraints();
+        setLayout(new MigLayout("", "[100:pref,fill]", "[100:pref,fill]"));
 
-        setLayout(new GridBagLayout());
+        mJailPanel.addComponentListener(new ComponentListener() {
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/TeamStatusPanel.java]: This is adding a `ComponentListener` to `mJailPanel` which calls another method on `mJailPanel`....can the listener just go directly IN `JailPanel`?

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/frame/ChessFrame.java [hunk 3]
```diff
@@ -74,6 +76,20 @@ public void actionPerformed(ActionEvent e) {
             }
         });
 
+        setPreferredSize(new Dimension(685, 450));
```
Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/frame/ChessFrame.java]: Let's not add this stuff to `ChessFrame` and let each child frame size itself

File: ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java (hunk 1 above)
Comment (ghost) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java]: Checking the Mod to see if there is remainder or not. If there is remainder we add an additional row of 4.

Comment (drewhannay) [inline ChessCrafterDesktop/src/main/java/com/drewhannay/chesscrafter/panel/JailPanel.java]: Sorry, I meant use `JAIL_WIDTH` instead of `4`

**Pull request merged**